src/adagrams.js

```diff
@@ -1,15 +1,163 @@
 export const drawLetters = () => {
-  // Implement this method for wave 1
+  const letterPool = {
+    A: 9,
+    B: 2,
+    C: 2,
+    D: 4,
+    E: 12,
+    F: 2,
+    G: 3,
+    H: 2,
+    I: 9,
+    J: 1,
+    K: 1,
+    L: 4,
+    M: 2,
+    N: 6,
+    O: 8,
+    P: 2,
+    Q: 1,
+    R: 6,
+    S: 4,
+    T: 6,
+    U: 4,
+    V: 2,
+    W: 2,
+    X: 1,
+    Y: 2,
+    Z: 1,
+  };
+
+  let letters = "";
+  for (const [letter, num] of Object.entries(letterPool)) {
+    letters += letter.repeat(num);
+  }
+
+  const randomlyHand = [];
+  const lettersCountDict = {
+    A: 0,
+    B: 0,
+    C: 0,
+    D: 0,
+    E: 0,
+    F: 0,
+    G: 0,
+    H: 0,
+    I: 0,
+    J: 0,
+    K: 0,
+    L: 0,
+    M: 0,
+    N: 0,
+    O: 0,
+    P: 0,
+    Q: 0,
+    R: 0,
+    S: 0,
+    T: 0,
+    U: 0,
+    V: 0,
+    W: 0,
+    X: 0,
+    Y: 0,
+    Z: 0,
+  };
+
+  while (randomlyHand.length < 10) {
+    const i = Math.floor(Math.random() * letters.length);
+    const selectedLetter = letters[i];
+
+    if (lettersCountDict[selectedLetter] < letterPool[selectedLetter]) {
+      lettersCountDict[selectedLetter]++;
+      randomlyHand.push(selectedLetter);
+    }
+  }
+
+  return randomlyHand;
 };
 
 export const usesAvailableLetters = (input, lettersInHand) => {
-  // Implement this method for wave 2
+  let lettersInHandCopy = [];
+  let wordLower = input.toLowerCase();
+
+  for (let letter of lettersInHand) {
+    lettersInHandCopy.push(letter.toLowerCase());
+  }
+
+  for (let letter of wordLower) {
+    if (!lettersInHandCopy.includes(letter)) {
+      return false;
+    } else {
+      lettersInHandCopy.splice(lettersInHandCopy.indexOf(letter), 1);
```

Comment:
`splice` like python's `remove` has linear time complexity with respect to the length of the list. This won't be a problem for the size of data we're working with, but consider using a strategy of virtually "dividing" the list into a used and unused side. Swapping a value to the used side would be constant time!